gdk4/src/rgba.rs

```diff
@@ -67,15 +67,49 @@ impl RGBABuilder {
 
 impl RGBA {
     #[inline]
-    pub fn new(red: f32, green: f32, blue: f32, alpha: f32) -> Self {
+    pub const fn new(red: f32, green: f32, blue: f32, alpha: f32) -> Self {
         skip_assert_initialized!();
-        unsafe {
-            Self::unsafe_from(ffi::GdkRGBA {
+        Self {
+            inner: ffi::GdkRGBA {
                 red,
                 green,
                 blue,
                 alpha,
-            })
+            },
+        }
+    }
+
+    #[inline]
+    pub const fn with_red(self, red: f32) -> Self {
+        Self {
+            inner: ffi::GdkRGBA { red, ..self.inner },
+        }
```

Comment:
just call self.set_red(red); and return self?

Reply:
That's possible, but it can't be const since `set_red` can't be const because it takes `&mut self`

Reply:
I am fine with either

Reply:
As in, the latter is not clear that it is actually copying the struct, and not mutating